stellargraph/mapper/full_batch_generators.py

```diff
@@ -379,10 +309,14 @@ def __init__(self, G, name=None, sparse=True, transform=None):
         for edge_type in edge_types:
 
             col_index = [
-                node_index[n1] for n1, n2, etype in G.edges if etype == edge_type
+                self.node_index[n1]
+                for n1, n2, etype in G.edges(triple=True)
+                if etype == edge_type
             ]
             row_index = [
-                node_index[n2] for n1, n2, etype in G.edges if etype == edge_type
+                self.node_index[n2]
+                for n1, n2, etype in G.edges(triple=True)
+                if etype == edge_type
             ]
```

Comment:
Perhaps this functionality could be something for the new StellarGraph class?  Also have a look at HinSAGE and the `GraphSchema` class which was supposed to be a place for this functionality.  *There is no need to change this PR* but perhaps in the  future these operations can be done in the StellarGraph class so they can be optimized.